### ISSUE TITLE
regression transform not correct when axis in log scale

### ISSUE THREAD

**Issue by robna** [altair-viz/altair#3225]
When using `transform_regression(...)` (linear), while one axis is on log scale, i.e. `x=alt.X('x', scale=alt.Scale(type='log'))` I get something like the plot below with the blue regression line. It should however be like the red (hand drawn) line shows....

Altair version is 5.01, but I think this is not new.

![visualization_RegTest](https://github.com/altair-viz/altair/assets/26489593/ff8e271c-4c19-41ee-80c1-0dd91986760d)

**Comment by ChristopherDavisUCI** [comment]
Thanks for the report!  To me at first glance this seems like a Vega-Lite issue: [Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4Igxg9gdgZglgcxALlANzgUwO4tJKAFzigFcJSBnAdTgBNCALFAZgAY2AacaYsiygAlMiRoVYcAvpO4AbAIYBPTACcUAbVABbeSoDWeEIUUAHTChAmIJcTJCYokOiSSoQAD0PxMsuhcEQKpSYVtiqINyUYPKy5m7GZhayEEh2CXEgAI6k8kRwhPLEaOZ2il5YvhYAsnCxlAD6Zir1AOIxyVARRqYZ2bnEBUUl0pzaugbxPUkkJdwOTi6Gnm7elcggAUEhEGFqkdGxhulJKSBpU+t9eYNwxWfcZSsVfus1dY2qre3QXceXOddCrdhtxCCpcpQYIEtBpQD91ptgqFwtwVJgEGjKJQ4PCQG9MA0ml9ZB0zgBdSRk7h0QryQxQeRaDI0goAWgATABWOh0dkAFk5MDYmDAAA52QA2ACMnIlYCl7HZUolAHZ+QBOTAwdkse4gAAkUUYmB0FjEhBMlGQAHprcUEPIAHQIfKMUgAI0dOOtRpN8jt6PkrNk+Uwds5jplkcdACtKD9qbTgoRKIYWUGuTz+YLhWLJTK5Qq2ErVRqtTrYRtAkiduFkBLuPjCZ82iT4aKZKBEdtdigpY3agSPs1W6TkHzO1Wtsi1OOB+8iaP4RLJ92Zyg+fOh4vvp1kCxKdIgA).  I tried searching Vega-Lite issues but didn't see anything.  (Is there any chance this is intended behavior, with the line of best fit being a line with respect to the chosen scales?  That doesn't seem correct to me, but I don't work much with log scales...)

**Comment by robna** [comment]
Yes, apparently when using linear regression, Vega-Lite represents the line is only represented by the first and last coordinate. Connecting them will result in a straight line also when the axis is on log scale.

Other regression methods, that may result in a curved line (like `pow`) will include intermediate points even if the relation is a perfectly straight line and optimally approximated by $y = x^1$.

This can be seen when switching the regression mark to circle.

```python
import altair as alt
import pandas as pd
import numpy as np

np.random.seed(0)
df = pd.DataFrame({
    'x': [1, 3, 5, 7],
    'y': [1, 3, 5, 7],
})

scatter = alt.Chart(df).mark_point(size=400).encode(
    x=alt.X('x', scale=alt.Scale(type='log')),
    y='y'
)
reg_line = scatter.transform_regression('x', 'y', method='pow').mark_circle(size=100, color='red', )

scatter + reg_line
```
![image](https://github.com/altair-viz/altair/assets/26489593/1ab5bb0e-1221-47a0-9886-88ae160ab245)

**Comment by ChristopherDavisUCI** [comment]
Thanks @robna for investigating this further!  Would it make sense to open an issue with Vega-Lite? https://github.com/vega/vega-lite/issues

**Comment by jardelva96** [comment]
Regression Transform Not Correct When Axis in Log Scale
The problem you're encountering involves using the transform_regression function in Altair while having one of the axes in log scale. The issue is that the linear regression line does not behave as expected in log-transformed scales, and this results in a misleading representation of the data.
When plotting a regression line on a log-scaled axis, the line should ideally account for the logarithmic transformation and appear curved to accurately represent the relationship between the variables.
this can be solve the issue

import pandas as pd
import numpy as np
import altair as alt

#Create a dummy DataFrame with random data
np.random.seed(0)
df = pd.DataFrame({
    'Horsepower': np.random.randint(1, 10, size=4),
    'Miles_per_Gallon': np.random.randint(1, 10, size=4)
})

#Create a scatter chart
scatter = alt.Chart(df).mark_point().encode(
    x=alt.X('Horsepower:Q', scale=alt.Scale(type='log')),
    y='Miles_per_Gallon:Q'
)

#Create a linear regression line overlay
reg_line = scatter.transform_regression('Horsepower', 'Miles_per_Gallon').mark_line()

#Display the chart
chart = scatter + reg_line
chart



When using transform_regression with a log scale, Vega-Lite computes the regression line based on the original (non-transformed) data. The line connects only the first and last coordinates, resulting in a straight line, even on a log scale. This behavior doesn't account for the log transformation, which is why the result is unexpected.
